### DISCUSSION TITLE
My XIAOMI5 phone can't use this app

### DISCUSSION THREAD

**Discussion by TFireOps** [Mygod/VPNHotspot#224] [Q&A] [answered]
Hello author
I like your app very much, but when I use it on my Mi5, it does not work properly. When my computer is connected to the hotspot of the mobile phone, it still passes through the network of the mobile phone instead of the VPN tunnel.
Although in the Google App Store, I saw some people say that setting the IP blocking mode to None is enough, but I did it, and it didn’t work properly.
By the way, there is no network sharing hardware acceleration in the developer mode. Android version is 8
So do you have a way to fix this bug. Thanks~

Attach log
[vpnhotspot-1091469617707688154.log](https://github.com/Mygod/VPNHotspot/files/6459864/vpnhotspot-1091469617707688154.log)

**Comment by TFireOps** [comment]
MIUI10.2

**Comment by Mygod** [comment]
Turn off hardware tethering acceleration.

**Reply by TFireOps** [reply]
Hello, author.
I know you said turning off hardware tethering acceleration..
But my Mi 5 phone does not have this option.
I have used the latest official Mi 5 system.
But it still cannot be used.
I have looked for it in the developer options, there is no such option

So I want to ask.
If there is no hardware tethering acceleration. option.
How can I turn off this option?

**Reply by Mygod** [reply]
Sorry my bad. Could you export debug information while the service is running?

**Reply by TFireOps** [reply]
ok,This is a new log
[vpnhotspot-4199453108585895523.log](https://github.com/Mygod/VPNHotspot/files/6462829/vpnhotspot-4199453108585895523.log)

**Reply by Mygod** [reply]
Bruh how in the world does your device have hardware tethering acceleration support on Android 8.0??

**Reply by Mygod** [reply]
(For context, this feature was added to AOSP only after Android 8.1.)
https://android.googlesource.com/platform/frameworks/base/+/5a7dea1a8eb2cf53fc1d5a52004647de94150e62%5E%21/#F1

**Reply by TFireOps** [reply]
so I can't use this app in this system version?

**Reply by TFireOps** [reply]
android8.0.0

**Comment by Mygod** [comment, marked as answer]
Please try this build: https://www108.zippyshare.com/v/p8qbVKR9/file.html

Go to app settings and see if there is an option for turning off tethering acceleration.

**Reply by TFireOps** [reply]
Oh! There is a network sharing hardware acceleration option in the app settings now, and after I closed it, my computer can also access the Internet through the VPN channel! Thank you, maintainer!
